scripts/RelayHubGasEstimation.ts

```diff
@@ -0,0 +1,641 @@
+    token = await deployContract<UtilToken>('UtilToken')
+    const smartWalletTemplate = await deployContract<SmartWallet>('SmartWallet')
+  
+    relayHub = await deployRelayHub(penalizer.address)
+  
+    await fundedAccount.sendTransaction({
+      to: owner.address,
+      value: ethers.utils.parseEther('1'),
+    })
+  
+    const factory = (await createSmartWalletFactory(
+      smartWalletTemplate,
+      false,
+      owner,
+    )) as SmartWalletFactory
+  
+    forwarder = (await createSupportedSmartWallet({
+      relayHub: relayHubSigner.address,
+      factory,
+      owner,
+      sender: relayHubSigner,
+    })) as SmartWallet
+  
+    await token.mint('1000', forwarder.address)
+  
+    relayRequest = {
+      request: {
+        relayHub: relayHub.address,
+        to: recipient.address,
+        data: '0xdeadbeef',
+        from: owner.address,
+        nonce: (await forwarder.nonce()).toString(),
+        value: '0',
+        gas: '3000000',
+        tokenContract: token.address,
+        tokenAmount: '1',
+        tokenGas: '50000',
+        validUntilTime: '0',
+      },
+      relayData: {
+        gasPrice,
+        feesReceiver: relayWorker.address,
+        callForwarder: forwarder.address,
+        callVerifier: verifier.address,
+      },
+    }
+  
+    await relayHub
+      .connect(relayOwner)
+      .stakeForAddress(relayManager.address, 1000, {
+        value: ethers.utils.parseEther('2'),
+      })
+  
+    await relayHub.connect(relayManager).addRelayWorkers([relayWorker.address])
+    await relayHub.connect(relayManager).registerRelayServer('http://relay.com')
+
+    return {
+      forwarder, recipient, relayHub, owner, relayRequest
+    }
+  }
+
+  async function printGasStatus(receipt: ContractReceipt) {
+    const [firstAccount] = (await ethers.getSigners()) as [SignerWithAddress]
+    await token.mint('1000', firstAccount.address)
+
+    const noRelayCall = await token.transfer(owner.address, '1000')
+    const { gasUsed: gasUsedWithoutRelay } = await noRelayCall.wait()
+
+    const gasOverhead = receipt.gasUsed.sub(gasUsedWithoutRelay)
+    console.log(
+      `Destination Call Without enveloping - Gas Used: ${gasUsedWithoutRelay.toString()}`,
+    )
+    console.log(
+      `Destination Call with enveloping - Gas Used: ${receipt.gasUsed.toString()}`,
+    )
+    logGasOverhead(gasOverhead)
+  }
+
+  function printGasPrediction({
+    internalDestinationCallCost,
+    internalTokenCallCost,
+  }: {
+    internalDestinationCallCost: number;
+    internalTokenCallCost: number;
+  }) {
+    // gas estimation fit
+    // 5.10585×10^-14 x^3 - 2.21951×10^-8 x^2 + 1.06905 x + 92756.3
+    // 8.2808×10^-14 x^3 - 8.62083×10^-8 x^2 + 1.08734 x + 36959.6
+    const a0 = Number('35095.980');
+    const a1 = Number('1.098');
+    const estimatedCost =
+      a1 * (internalDestinationCallCost + internalTokenCallCost) + a0;
+    console.log('The predicted total cost is: ', estimatedCost);
+  }
```

Comment:
I'd remove this function because:
1. this estimation is quite different from the real cost
2. the values used in the [client](https://github.com/rsksmart/rif-relay-client/blob/main/src/gasEstimator/utils.ts#L93) are different from those used here